docs/mcap/README.md

```diff
@@ -0,0 +1,371 @@
+remote.
+
+#### Listing and accessing attachments
+The format provides the ability to list attachments contained wihtin the file,
+and quickly extract them from the file contents. To list/select attachments
+in the file:
+1. Read the fixed-length footer and seek to the start of the index data
+   section.
+2. Scan forward until encountering the attachment index, then read attachment
+   index records until encountering a record that is not an attachment index.
+3. The rcords covered in the previous read will include attachment names,
+   types, sizes, and timestamps. These can be used to fill out a list of
+   attachments for selection.
+4. To select an attachment from th efile, seek to the associated offset in the
+   file and unpack the file content from the attachment record.
+
+#### Accessing summary statistics
+The format provides for fast local or remote access to summary information in
+the same style as "rosbag info", with the intent of functional parity with
+rosbag info. For reference, here is an example of the rosbag info output:
+
+```
+path:         demo.bag
+version:      2.0
+duration:     7.8s
+start:        Mar 21 2017 19:26:20.10 (1490149580.10)
+end:          Mar 21 2017 19:26:27.88 (1490149587.88)
+size:         67.1 MB
+messages:     1606
+compression:        lz4 [79/79 chunks; 56.23%]
+uncompressed:       119.1 MB @ 15.3 MB/s
+compressed:    67.0 MB @  8.6 MB/s (56.23%)
+types:        diagnostic_msgs/DiagnosticArray [60810da900de1dd6ddd437c3503511da]
+              radar_driver/RadarTracks        [6a2de2f790cb8bb0e149d45d297462f8]
+              sensor_msgs/CompressedImage     [8f7a12909da2c9d3332d540a0977563f]
+              sensor_msgs/PointCloud2         [1158d486dd51d683ce2f1be655c3c181]
+              sensor_msgs/Range               [c005c34273dc426c67a020a87bc24148]
+              tf2_msgs/TFMessage              [94810edda583a504dfda3829e70d7eec]
+topics:        /diagnostics               52 msgs    : diagnostic_msgs/DiagnosticArray
+              /image_color/compressed       234 msgs    : sensor_msgs/CompressedImage    
+              /radar/points             156 msgs    : sensor_msgs/PointCloud2        
+              /radar/range              156 msgs    : sensor_msgs/Range              
+              /radar/tracks             156 msgs    : radar_driver/RadarTracks       
+              /tf                       774 msgs    : tf2_msgs/TFMessage             
+              /velodyne_points           78 msgs    : sensor_msgs/PointCloud2
+```
+
+The reader will recover this data from the index as follows:
+
+The reader will recover this data from the index as follows:
+1. Read the fixed length footer and seek to the index_offset.
+2. Read the run of channel info records that follow to get topic names, types,
+and MD5 data (which in case of ROS1 will be in the user data section), as well
+as channel IDs to interpret the chunk index records.
```

Comment:
I don't really like how much ROS stuff we are putting in this spec doc. Maybe this "feature explanations" section should be a separate doc?

Reply:
split this into a separate doc, and linked from the "further reading" section.